src/calendar/header/index.tsx

```diff
@@ -72,116 +70,101 @@ export interface CalendarHeaderProps {
   importantForAccessibility?: 'auto' | 'yes' | 'no' | 'no-hide-descendants';
 }
 
-class CalendarHeader extends Component<CalendarHeaderProps> {
-  static displayName = 'CalendarHeader';
-
-  static propTypes = {
-    month: PropTypes.instanceOf(XDate),
-    addMonth: PropTypes.func,
-
-    theme: PropTypes.object,
-    firstDay: PropTypes.number,
-    displayLoadingIndicator: PropTypes.bool,
-    showWeekNumbers: PropTypes.bool,
-    monthFormat: PropTypes.string,
-    hideDayNames: PropTypes.bool,
-    hideArrows: PropTypes.bool,
-    renderArrow: PropTypes.func,
-    onPressArrowLeft: PropTypes.func,
-    onPressArrowRight: PropTypes.func,
-    disableArrowLeft: PropTypes.bool,
-    disableArrowRight: PropTypes.bool,
-    disabledDaysIndexes: PropTypes.arrayOf(PropTypes.number),
-    renderHeader: PropTypes.any,
-    customHeaderTitle: PropTypes.any,
-    
-    webAriaLevel: PropTypes.number
-  };
-
-  static defaultProps = {
-    monthFormat: 'MMMM yyyy',
-    webAriaLevel: 1
-  };
-  style: any;
-
-  constructor(props: CalendarHeaderProps) {
-    super(props);
-
-    this.style = styleConstructor(props.theme);
-  }
-
-  shouldComponentUpdate(nextProps: CalendarHeaderProps) {
-    if (!sameMonth(nextProps.month, this.props.month)) {
-      return true;
-    }
-    return shouldUpdate(this.props, nextProps, [
-      'displayLoadingIndicator',
-      'hideDayNames',
-      'firstDay',
-      'showWeekNumbers',
-      'monthFormat',
-      'renderArrow',
-      'disableArrowLeft',
-      'disableArrowRight',
-      'renderHeader',
-      'customHeaderTitle'
-    ]);
-  }
-
-  addMonth = () => {
-    const {addMonth} = this.props;
-    addMonth?.(1);
+const CalendarHeader = forwardRef((props: CalendarHeaderProps, ref) => {
+  const {
+    theme,
+    style: propsStyle,
+    addMonth: propsAddMonth,
+    month,
+    monthFormat,
+    firstDay,
+    hideDayNames,
+    showWeekNumbers,
+    hideArrows,
+    renderArrow,
+    onPressArrowLeft,
+    onPressArrowRight,
+    disableArrowLeft,
+    disableArrowRight,
+    disabledDaysIndexes,
+    displayLoadingIndicator,
+    customHeaderTitle,
+    renderHeader,
+    webAriaLevel,
+    testID,
+    accessibilityElementsHidden,
+    importantForAccessibility
+  } = props;
+  const style = styleConstructor(theme);
```

Comment:
use ref/state for this, otherwise you creating a new object each render

Reply:
Thanks. Done